File: 프로그래머스/unrated/181839. 주사위 게임 1/주사위 게임 1.java
class Solution {
    public int solution(int a, int b) {
        int answer = 0;
        boolean aa = (a%2!=0);
        boolean bb = (b%2!=0);
        
        if (aa && bb) answer = a*a + b*b;
        else if (aa || bb) answer = 2*(a+b);
        else answer= Math.abs(a-b);
        
        return answer;
    }
}

File: 프로그래머스/unrated/181839. 주사위 게임 1/주사위 게임 1.py
import math

def solution(a, b):
    answer = 0
    if (a*b)%2!=0:
        answer = a*a+b*b
    else:
        if (a%2==0) and (b%2==0):
            answer = abs(a-b)
        else:
            answer = 2*(a+b)
    return answer